core/tests/protractor_utils/TopicEditorPage.js

```diff
@@ -150,10 +156,40 @@ var TopicEditorPage = function() {
     });
   };
 
+  this.expectNumberOfQuestionsForSkillWithIndexToBe = function(
+      count, index) {
+    selectSkillDropdown.click();
+    element.all(by.tagName('option')).then(
+      function(options) {
+        options[index + 1].click();
+      });
+    waitFor.visibilityOf(
+      questionItem, 'Question takes too long to appear');
+    questionItems.then(function(items) {
+      expect(items.length).toEqual(count);
+    });
+  };
+
   this.saveQuestion = function() {
+    waitFor.elementToBeClickable(
```

Comment:
Thanks for adding this, can we do something to validate the click has worked as expected?